### PR TITLE
Hasura Graphql integration and refactor

### PR DESCRIPTION
**Pull request by sashaboi**
Jiska Darr tha wohi kiya : Refactor. 
Moving towards prod , having readable , easy to navigate code is important , so refactored code from just a lonely app.js file, to a complete folderstructure. 

Also added router ,and apollo client for using Graphql from Hasura ( read : Backend for maintaining leaderboard)

Yet to do : styling properly, figuring out how to get user info properly. 
This is project flow ( I think for now )
https://whimsical.com/card-game-62cxdPkWktrfUAJvwhYjYd@7YNFXnKbYkKDwbrD1Azak

PREVIEW : https://magenta-mooncake-426b57.netlify.app/
Please review !

### CODE REVIEW

**Review by bhavyaj12 — APPROVED** [review]
Great work!
Nice that you have added different routes, now this will be a complete application. Couldn't really find any significant changes to suggest, minor spacing and formatting issues you can take care of.
Great that you have explored graphql! Nice workflow too, leveling up everything I see :)

Oh and deploy preview isn't working for some reason, to check that

File: src/contexts/graphql-apollo-provider.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+const errorLink = onError(({graphqlErrors,networkError})=>{
+  if(graphqlErrors){
+    graphqlErrors.forEach(({message,location,path})=>{
+      alert(`Graphql Error: ${message}`)
+    })
+  }
+})
+
+const urltohit = from([
```
Comment (athar05) [inline src/contexts/graphql-apollo-provider.js:12]: This is improper variable naming convention. I'm sure you know this, but just pointing it out anyway.

File: src/styles.css [hunk 2]
```diff
@@ -1,30 +1,103 @@
...
   grid-template-columns: repeat(4, 1fr);
   grid-template-rows: repeat(4, 1fr);
 }
 .card-component-btn{
-  padding: 2rem;
+  height: 10vh;
+  width: 10vh;
+  padding: 1rem;
   font-size: 2rem;
-  background-color: #4d1635;
```
Comment (athar05) [inline src/styles.css:16]: You can consider using variables.

**Review by athar05 — COMMENTED** [review]
Your code looks clean and good, hence I have taken the liberty to nitpick here and there. Amazing job

**Comment by athar05** [comment]
In terms of functionality, the app is really good. Although I feel there is scope for improvement in terms of styling, but I understand if that's not the priority. Also, it's great to see that you have used GraphQL, motivates me to explore it as well. 

Kudos man, amazing job.

**Pull request merged**